featurebyte/session/databricks_unity.py

```diff
@@ -129,3 +176,39 @@ async def list_tables(
             for _, (name,) in tables[["TABLE_NAME"]].iterrows():
                 output.append(TableSpec(name=name))
         return output
+
+    async def create_table_as(
+        self,
+        table_details: TableDetails | str,
+        select_expr: Select | str,
+        kind: Literal["TABLE", "VIEW"] = "TABLE",
+        partition_keys: list[str] | None = None,
+        replace: bool = False,
+        retry: bool = False,
+        retry_num: int = 10,
+        sleep_interval: int = 5,
+    ) -> pd.DataFrame | None:
+        result = await super().create_table_as(
+            table_details,
+            select_expr,
+            kind,
+            partition_keys,
+            replace,
+            retry,
+            retry_num,
+            sleep_interval,
+        )
+
+        # grant ownership of the table to the group
+        if isinstance(table_details, str):
+            table_details = TableDetails(
+                database_name=None, schema_name=None, table_name=table_details
+            )
+        fully_qualified_table_name = sql_to_string(
+            get_fully_qualified_table_name(table_details.dict()), source_type=self.source_type
+        )
+        grant_permissions_query = (
+            f"ALTER {kind} {fully_qualified_table_name} OWNER TO `{self.group_name}`"
+        )
+        await self.execute_query(grant_permissions_query)
```

Comment:
Would it be useful to have a helper method to alter owner? Would be easier to make changes later if it is in a centralised place.